sqlalchemy-stubs/dialects/postgresql/hstore.pyi

```diff
@@ -0,0 +1,58 @@
+from ... import types as sqltypes
+from ...sql import functions as sqlfunc, type_api
+from .array import ARRAY
+from typing import Any, Optional, Mapping, Type, List, Text
+
+class HSTORE(sqltypes.Indexable, sqltypes.Concatenable, sqltypes.TypeEngine[Mapping[str, Any]]):
+    __visit_name__: str = ...
+    hashable: bool = ...
+    text_type: type_api.TypeEngine[Text] = ...
+    def __init__(self, text_type: Optional[type_api.TypeEngine[Text]] = ...) -> None: ...
+    class Comparator(type_api.TypeEngine.Comparator):
+        def has_key(self, other: Any): ...
+        def has_all(self, other: Any): ...
+        def has_any(self, other: Any): ...
+        def contains(self, other: Any, **kwargs: Any): ...
+        def contained_by(self, other: Any): ...
+        def defined(self, key: Any): ...
+        def delete(self, key: Any): ...
+        def slice(self, array: Any): ...
+        def keys(self): ...
+        def vals(self): ...
+        def array(self): ...
+        def matrix(self): ...
+    comparator_factory: Type[Any] = ...
+    def bind_processor(self, dialect: Any): ...
+    def result_processor(self, dialect: Any, coltype: Any): ...
+
+class hstore(sqlfunc.GenericFunction[Mapping[str, Any]]):
+    type: HSTORE = ...
+    name: str = ...
+
+class _HStoreDefinedFunction(sqlfunc.GenericFunction[bool]):
+    type: sqltypes.Boolean = ...
+    name: str = ...
+
+class _HStoreDeleteFunction(sqlfunc.GenericFunction[Mapping[str, Any]]):
+    type: HSTORE = ...
+    name: str = ...
+
+class _HStoreSliceFunction(sqlfunc.GenericFunction[Mapping[str, Any]]):
+    type: HSTORE = ...
+    name: str = ...
+
+class _HStoreKeysFunction(sqlfunc.GenericFunction[List[Text]]):
+    type: ARRAY[Text] = ...
+    name: str = ...
+
+class _HStoreValsFunction(sqlfunc.GenericFunction[List[Text]]):
+    type: ARRAY[Text] = ...
+    name: str = ...
+
+class _HStoreArrayFunction(sqlfunc.GenericFunction[List[Text]]):
+    type: ARRAY[Text] = ...
+    name: str = ...
+
+class _HStoreMatrixFunction(sqlfunc.GenericFunction[List[Text]]):
```

Comment:
Just to double-check: do we really need al these underscored (private?) classes?

Reply:
I didn't fill out all of the return types for the comparator functions which is where these are used. I'll do that now.